Source/Tests/LinuxSyscalls/FileManagement.cpp

```diff
@@ -313,6 +278,44 @@ FileManager::FileManager(FEXCore::Context::Context *ctx)
     }
   }
 
+  // Now that we loaded the thunks object, walk through and ensure dependencies are enabled as well.
+  for (auto const &DBObject : ThunkDB) {
+    if (DBObject.second.Enabled) {
+      auto ThunkPath = ThunkGuestPath / DBObject.second.LibraryName;
+      if (std::filesystem::exists(ThunkPath)) {
+        for (auto Overlay : DBObject.second.Overlays) {
+          // Direct full path in guest RootFS to our overlay file
+          ThunkOverlays.emplace(Overlay, ThunkPath);
+        }
+      }
+
+      // Now walk the dependencies and set them up as well
+      // Make sure to enable each one as we go to remove circular dependencies
+      std::function<void(const std::unordered_set<std::string> &Depends)> InsertDependencies
+        = [this, &ThunkGuestPath, &InsertDependencies](const std::unordered_set<std::string> &Depends) -> void {
+        for (auto const &Depend : Depends) {
+          auto DBDepend = ThunkDB.find(Depend);
+          if (DBDepend != ThunkDB.end() &&
+              DBDepend->second.Enabled == false) {
+
+            auto ThunkPath = ThunkGuestPath / DBDepend->second.LibraryName;
```

Comment:
This block is now duplicated outside the loop. Can we instead call `InsertDependencies({DBObject.first})` at the top level? That should make the loop body a tad more organized, and the extra lookup should be cheap.

Reply:
Resolved.